lib/manager/gradle-lite/parser.ts

```diff
@@ -0,0 +1,294 @@
+
+    if (matcher.tokenMapKey) {
+      result[matcher.tokenMapKey] = token;
+    }
+  }
+
+  tokens.splice(0, matchers.length - lookaheadCount);
+  return result;
+}
+
+const matcherConfigs: SyntaxMatchConfig[] = [
+  {
+    // foo = 'bar'
+    matchers: [
+      { matchType: TokenType.Word, tokenMapKey: 'keyToken' },
+      { matchType: TokenType.Assignment },
+      { matchType: TokenType.String, tokenMapKey: 'valToken' },
+      {
+        // Ensure we skip assignments of complex expressions (not strings)
+        matchType: [
+          TokenType.Colon,
+          TokenType.RightBrace,
+          TokenType.Word,
+          TokenType.String,
+          TokenType.StringInterpolation,
+        ],
+        lookahead: true,
+      },
+    ],
+    handler: function handleAssignment({
+      packageFile,
+      tokenMap,
+    }: SyntaxHandlerInput): SyntaxHandlerOutput {
+      const { keyToken, valToken } = tokenMap;
+      const variableData: VariableData = {
+        key: keyToken.value,
+        value: valToken.value,
+        fileReplacePosition: valToken.offset,
+        packageFile,
+      };
+      return { vars: { [variableData.key]: variableData } };
+    },
+  },
+  {
+    // 'foo.bar:baz:1.2.3'
+    matchers: [
+      {
+        matchType: TokenType.String,
+        tokenMapKey: 'token',
+      },
+    ],
+    handler: function processDepString({
+      packageFile,
+      tokenMap,
+    }: SyntaxHandlerInput): SyntaxHandlerOutput {
+      const { token } = tokenMap;
+      const dep = parseDependencyString(token.value);
+      if (dep) {
+        dep.managerData = {
+          fileReplacePosition: token.offset + dep.depName.length + 1,
+          packageFile,
+        };
+        return { deps: [dep] };
+      }
+      return null;
+    },
+  },
+  {
+    // "foo.bar:baz:${bazVersion}"
+    matchers: [
+      {
+        matchType: TokenType.StringInterpolation,
+        tokenMapKey: 'depInterpolation',
+      },
+    ],
+    handler: function processDepInterpolation({
+      tokenMap,
+      variables,
+    }: SyntaxHandlerInput): SyntaxHandlerOutput {
+      const token = tokenMap.depInterpolation as StringInterpolation;
+      const interpolationResult = interpolateString(token.children, variables);
+      if (interpolationResult && isDependencyString(interpolationResult)) {
+        const dep = parseDependencyString(interpolationResult);
+        if (dep) {
+          const lastChild = token.children[token.children.length - 1];
+          const lastChildValue = lastChild?.value;
+          const variable = variables[lastChildValue];
+          if (
+            lastChild?.type === TokenType.Variable &&
+            variable &&
+            variable?.value === dep.currentValue
+          ) {
+            dep.managerData = {
+              fileReplacePosition: variable.fileReplacePosition,
+              packageFile: variable.packageFile,
+            };
+            dep.groupName = variable.key;
```

Comment:
Ref: #6474

Reply:
The potential problem here: same variable names in different files.
Possible solution: use `packageFile` in group name.